### PR TITLE
Template Parts: Fix modal search stacking context

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
PR fixes the stacking context of the Search component in the "Choose template part" modal.

## Why?
I noticed that I couldn't focus on searching while scrolling through the template parts. Clicking on the search field was selecting the template part.

## How?
Updating the `z-index` to 2 seems to resolve the issue.

## Testing Instructions
1. Open the Site Editor.
2. Insert Template Part.
3. Click the "Choose" button in the placeholder.
4. Scroll through the results.
5. Confirm that the search field is focusable.

## Screenshots or screencast 

https://user-images.githubusercontent.com/240569/206655551-a50fbe9b-ae8a-4ecb-bcce-92f6335a4657.mp4

### CODE REVIEW

**Comment by ntsekouras** [comment]
Thanks for the PR George! Can you take a look to apply the fix in `replace` flow as well? Didn't debug but the problem remains there.

**Comment by Mamaduka** [comment]
@ntsekouras, the replacement action uses the same modal, so there is no need for additional changes.

**Review by jasmussen — APPROVED** [review]

**Comment by ntsekouras** [comment]
>@ntsekouras, the replacement action uses the same modal, so there is no need for additional changes.

Hm... For me it didn't work(Chrome and site editor).. Let me try again.

**Review by ntsekouras — APPROVED** [review]
That's so weird that previously it worked for one of them, but rebuild and works as expected. Thanks George and sorry for the noise!
